web-app/HeimScripts/heatmap/run.R

```diff
@@ -114,26 +114,35 @@ addStats <- function(df, sorting, ranking, max_rows) {
     #this is the case for more than 1 sample
     df <-
       applySorting(df,sorting)  # no need for sorting in one sample case, we do it here only
-    if ( twoSubsets ){
-      markerTable  <- getDEgenes(df)  # relies on columns being sorted,
-    }                                 # with subset1 first, this is because of the way
-                                      # design matrix is being constructed
+
+    # is it valid measurements ?
+    validLimmaMeasurements <- isValidLimmaMeasurements(measurements)
+
+    if (twoSubsets && validLimmaMeasurements) {
+      markerTable  <- getDEgenes(measurements)  # relies on columns being sorted,
+    }                                           # with subset1 first, this is because of the way
+                                                # design matrix is being constructed
 
 
     useLimma <- !is.function(rankingMethod)  # rankingMethod is either a function or a character "limma"
     if (useLimma  && !twoSubsets) {  # cannot use any of the limma methods for single subset.
       stop( paste("Illegal ranking method: ", ranking, " two subsets needed.") )
     }
-    if (useLimma ) {
+
+    if (useLimma && validLimmaMeasurements) {
       if (!ranking %in% colnames(markerTable) ) {
           stop(paste("Illegal ranking method selected: ", ranking) )
       }
       rankingScore <- markerTable[ranking]
-
-    } else {
+    } else if (!useLimma && validLimmaMeasurements) {
       rankingScore <-
         apply(measurements, 1, rankingMethod, na.rm = TRUE )  # Calculating
-    }                                                         # ranking per probe (per row)
+    } else {
+      rankingScore <- data.frame(SIGNIFICANCE=numeric()) # when differential expression rank criteria
```

Comment:
Wouldn't it be better to use stop() to throw exception when user tries to use a differential expression rank criteria when the data is insufficient to calculate it?
If you create the heatmap in this scenario the probes will not be sorted in any meaningful way - why show a result which makes no sense?